MetaMask/metamask-extension
# Adding Input component and updating TextFieldBase

## Explanation
Adding `Input` component as a base component for `TextFieldBase`. Based on [TextField Audit and Insight Report](https://www.figma.com/file/aGW8sk6X6Jf9ac0MRMD4kX/TextField-Audit?node-id=349%3A23&t=xHFfsVJIbeO0FOxP-1)

Will be updating more TextField related components in other PRs for brevity 

#17584 

## Screenshots/Screencaps

https://user-images.githubusercontent.com/8112138/217631063-ca46d708-5509-4359-8e16-630870044dc3.mov

## Manual Testing Steps

- Go to the latest storybook build of this PR
- Search `Input` in the search panel
- Go to `Input` story in **Component Library** 
- Check all controls work
- Check all stories pass accessibility 
- Check documentation in Docs tabs
- Check console for errors

## Pre-merge author checklist


## Pre-merge reviewer checklist
